JUPE7H/Logic/Board.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace JUPE7H.Logic;
+
+using System.Text.Json;
+using UI.Structs;
+
+internal sealed class Board{
+    public const sbyte MINE = -1;
+    public const sbyte FLAG = -2;
+    public const sbyte UNMARKED = 127;
+    
+    public readonly Difficulty Difficulty;
+    public readonly short Width;
+    public readonly short Height;
+    public GameStatus Status{ get; private set; }
+    public sbyte this[int x, int y] => _visibleTiles[x, y];
+    public int RemainingMines => _mineCount - _flagCount;
+
+    private readonly int _seed;
+    private readonly int _mineCount;
+    private readonly sbyte[,] _realTiles;
+    private readonly sbyte[,] _visibleTiles;
+    private bool _isGenerated;
+    private int _flagCount;
+    private int _revealedCount;
+
+    private readonly SaveFile _saveFile;
+
+    public Board(Point size, Difficulty difficulty, int? seed = null){
+        Difficulty = difficulty;
+        Width = size.X;
+        Height = size.Y;
+        Status = GameStatus.InProgress;
+
+        _seed = seed ?? Guid.NewGuid().GetHashCode();
+        _mineCount = (int)(size.X * size.Y * (int)difficulty / 1000.0);
+        _realTiles = new sbyte[size.X, size.Y];
+        _visibleTiles = new sbyte[size.X, size.Y];
+        _isGenerated = false;
+        _flagCount = 0;
+        _revealedCount = 0;
+        
+        _saveFile = new SaveFile(new Point(Width, Height), Difficulty, _seed);
+
+        _visibleTiles.Fill(UNMARKED);
+    }
+
+    private void Generate(Point? safePosition = null){
+        _isGenerated = true;
+        Random random = new(_seed);
+        int placedMines = 0;
+
+        while(placedMines < _mineCount){
+            int x = random.Next() % Width;
+            int y = random.Next() % Height;
+            
+            if(_realTiles[x, y] == MINE) continue;
+            if(x == safePosition?.X && y == safePosition?.Y) continue;
+            
+            _realTiles[x, y] = MINE;
+            placedMines++;
+        }
+
+        for(int x = 0; x < Width; x++){
+            for(int y = 0; y < Height; y++){
+                _visibleTiles[x, y] = UNMARKED;
```

Comment:
`UNMARKED` gondolom egy konstans, ami ennyi erővel lehetett volna egy Enum is, ami C# esetén egy rendes típus.